src/xbot/common/subsystems/drive/control_logic/HeadingAssistModule.java

```diff
@@ -0,0 +1,64 @@
+package xbot.common.subsystems.drive.control_logic;
+
+import com.google.inject.assistedinject.Assisted;
+import com.google.inject.assistedinject.AssistedInject;
+
+import edu.wpi.first.wpilibj.Timer;
+import xbot.common.properties.DoubleProperty;
+import xbot.common.properties.XPropertyManager;
+import xbot.common.subsystems.pose.BasePoseSubsystem;
+
+public class HeadingAssistModule {
+
+    final BasePoseSubsystem pose;
+    final HeadingModule headingModule;
+    final DoubleProperty humanThreshold;
+    final DoubleProperty coastTime;
+    double desiredHeading;
+    double lastHumanInput;
+    boolean inAutomaticMode;
+    
+    @AssistedInject
+    public HeadingAssistModule(
+            @Assisted("headingModule") HeadingModule headingModule,
+            XPropertyManager propMan,
+            BasePoseSubsystem pose) {
+        this.headingModule = headingModule;
+        this.pose = pose;
+        humanThreshold = propMan.createPersistentProperty("HeadingAssistModule - Human Threshold", 0.05);
+        coastTime = propMan.createPersistentProperty("Heading Assist Module - Coast Time", 0.5);
+        lastHumanInput = 0;
+    }
+    
+    public double calculateHeadingPower(double humanRotationalPower) {
```

Comment:
It would be helpful to have a documentation comment with a reference implementation of calculating the rotational power param

Reply:
Excellent point. Including.